lib/diagnostic.rb

```diff
@@ -2,41 +2,46 @@
 
 # Ruby enumerable diagnostic
 class Diagnostic
+
   # Question 1
   # Write ruby code to return the longest entry in the list supplied.
   # Assume each entry in the list implements a length method that returns
   # a Fixnum.
   def longest(list)
-    # Your code goes here
+    array.select{|char_num| char_num.size.max}
   end
 
   # Question 2
   # Write ruby code to return the sum of entries in the
   # list supplied (i.e. apply the `+` operator).
   # Assume each entry implements a `+` method.
   def sum(list)
-    # Your code goes here
+    array.each(:+)
```

Comment:
`list` is the parameter so that is what you should be working with in the method:

`list.reduce(&:+)`